impl/src/main/java/org/jboss/cdi/tck/tests/lookup/dynamic/handle/InstanceHandleTest.java

```diff
@@ -0,0 +1,114 @@
+    @Test
+    public void testGetHandler() {
+        Client client = getContextualReference(Client.class);
+        BeanManager beanManager = getCurrentManager();
+
+        ActionSequence.reset();
+        assertNotNull(client);
+
+        Bean<?> alphaBean = beanManager.resolve(beanManager.getBeans(Alpha.class));
+        Instance<Alpha> instance = client.getAlphaInstance();
+
+        Instance.Handle<Alpha> alpha1 = instance.getHandle();
+        assertEquals(alphaBean, alpha1.getBean());
+        assertEquals(Dependent.class, alpha1.getBean().getScope());
+
+        String alpha2Id;
+
+        // Test try-with-resource
+        try (Instance.Handle<Alpha> alpha2 = instance.getHandle()) {
+            alpha2Id = alpha2.get().getId();
+            assertFalse(alpha1.get().getId().equals(alpha2Id));
+        }
+
+        List<String> sequence = ActionSequence.getSequenceData();
+        assertEquals(1, sequence.size());
+        assertEquals(alpha2Id, sequence.get(0));
+
+        alpha1.destroy();
+        // Subsequent invocations are no-op
+        alpha1.destroy();
+
+        sequence = ActionSequence.getSequenceData();
+        assertEquals(2, sequence.size());
+        assertEquals(alpha1.get().getId(), sequence.get(1));
+
+        // Test normal scoped bean is also destroyed
+        Instance<Bravo> bravoInstance = client.getInstance().select(Bravo.class);
+        String bravoId = bravoInstance.get().getId();
+        try (Instance.Handle<Bravo> bravo = bravoInstance.getHandle()) {
+            assertEquals(bravoId, bravo.get().getId());
+            ActionSequence.reset();
+        }
+        sequence = ActionSequence.getSequenceData();
+        assertEquals(1, sequence.size());
+        assertEquals(bravoId, sequence.get(0));
+    }
+
+    //@SpecAssertion(section = PROGRAMMATIC_LOOKUP, id = "TODO")
+    @Test
+    public void testHandlers() {
+        Instance<Processor> instance = getCurrentManager().createInstance().select(Processor.class);
+        ActionSequence.reset();
+        assertTrue(instance.isAmbiguous());
+        for (Instance.Handle<Processor> handler : instance.handles()) {
+            handler.get().ping();
+            if (handler.getBean().getScope().equals(Dependent.class)) {
+                handler.destroy();
+            }
+        }
+        assertEquals(3, ActionSequence.getSequenceSize());
+        ActionSequence.assertSequenceDataContainsAll("firstPing", "secondPing", "firstDestroy");
+
+        ActionSequence.reset();
+        assertTrue(instance.isAmbiguous());
+        for (Iterator<Instance.Handle<Processor>> iterator = instance.handles().iterator(); iterator.hasNext(); ) {
+            try (Instance.Handle<Processor> handler = iterator.next()) {
+                handler.get().ping();
+            }
+        }
+        assertEquals(4, ActionSequence.getSequenceSize());
+        ActionSequence.assertSequenceDataContainsAll("firstPing", "secondPing", "firstDestroy", "secondDestroy");
```

Comment:
How does ordering work here? I see `Processor` has `getPriority`, but that isn't called anywhere?

Reply:
@Ladicek it is undefined.
Which is why the assertion only checks for presence, not for ordering.

The `getPriority()` indeed isn't tied to `@Priority` anyhow. I copied the test from Weld and this seems to be a redundant method on the bean. I have already removed it.

Reply:
Ah gotcha! I'm not familiar with `ActionSequence` :-)